statsmodels/tsa/stattools.py

```diff
@@ -1182,6 +1183,147 @@ def _sigma_est_kpss(resids, nobs, lags):
+    using the updated 2010 tables. If the p-value is close to significant, then the critical values
+    should be used to judge whether to reject the null.
+
+    References
+    ----------
+    
+    Hamilton, J. D. 1994 .Time-series Analysis. Princeton: Princeton University Press.
+
+    Newey, W.K., and K.D. West.1987. A simple,postive semidefinite,heteroskedasticity and autocorrelation consistent
+    covariance matrix. Econometrica 55, 703-708.
+
+    Phillips, P. C. B., and P. Perron. 1988. Testing for a unit root in time series regression. Biometrika 75, 335-346.
+
+    P-Values (regression surface approximation) MacKinnon, J.G. 1994. “Approximate asymptotic distribution functions for 
+    unit-root and cointegration tests. Journal of Business and Economic Statistics 12, 167-76.
+
+    Critical values MacKinnon, J.G. 2010. “Critical Values for Cointegration Tests.” Queen’s University, Dept of Economics,
+    Working Papers. Available at http://ideas.repec.org/p/qed/wpaper/1227.html 
+
+    """
+
+    from warnings import warn
+
+    nobs = len(x)
+    x = np.asarray(x)
+
+    if regression not in ['c', 'nc', 'ct']:
+        raise ValueError("regression option %s not understood") % regression
+
+    if nobs != x.size:
+        raise ValueError("x of shape {0} not understood".format(x.shape))
+
+    hypo = regression.lower()
+
+    if lags is None:
+        # from Kwiatkowski et al. referencing Schwert (1989)
+        lags = int(np.ceil(12. * np.power(nobs / 100., 1 / 4.)))
+
+
+    rhs=x[:-1,None]
+    lhs=x[1:,None]
+
+    if hypo != 'nc' :
+        rhs = add_trend(rhs,hypo)
+
+    resols = OLS(lhs,rhs).fit()
+    k=rhs.shape[1]
+    n , u = resols.nobs , resols.resid
+    lam2 = _sigma_est_kpss(u,n,lags)
+    lam = np.sqrt(lam2)
+
+    s2 = u.dot(u) / (n - k) 
+    s = np.sqrt(s2)
+
+    gamma0 = s2 * (n - k) / n
+    sigma = resols.bse[0]
+    sigma2 = sigma ** 2.0
+    rho = resols.params[0]
+
+    stat_tau = np.sqrt(gamma0 / lam2) * ((rho - 1) / sigma ) - 0.5 * ((lam2 - gamma0) / lam) * (n * sigma / s)
+    stat_rho = n * (rho - 1) - 0.5 * (n ** 2.0 * sigma2 / s2) * (lam2 - gamma0)
+
+    nobs = int(resols.nobs)
+
+    if test_type == 'rho':
+        stat = stat_rho
+        dist_type = 'ADF-z'
+    else:
+        stat = stat_tau
+        dist_type = 'ADF-t'
+        
+
+    p_value = mackinnonp2(stat,regression=hypo,dist_type=dist_type)
+
+    crit = mackinnoncrit2(regression=hypo,nobs=n,dist_type=dist_type)
+
+    crit_dict = {'1%': crit[0], '5%': crit[1], '10%': crit[2]}
+
+    if store:
+        rstore = ResultsStore()
+        rstore.lags = lags
+        rstore.nobs = nobs
+
+        rstore.H0 = ("The coefficient on the lagged level equals 1 - "
+                       "unit root")
+        rstore.HA = "The coefficient on the lagged level < 1 - stationary"
+
+        return stat, p_value, crit_dict, rstore
+    else:
+        return stat, p_value, lags, crit_dict
```

Comment:
Can the multiple-return be merged? Something like `return stat, p_value, lags, crit_dict, rstore` and just set `rstore=None` in the latter case?